lib/service/service.go

```diff
@@ -2969,9 +2968,13 @@ type proxyListeners struct {
 	db            dbListeners
 	alpn          net.Listener
 	proxy         net.Listener
-	// gprc receives gRPC traffic that has the TLS ALPN protocol common.ProtocolProxyGRPC. This
-	// listener is only enabled when TLS routing is enabled.
-	grpc             net.Listener
+	// grpcInsecure receives gRPC traffic that has the TLS ALPN protocol common.ProtocolProxyGRPCInsecure. This
+	// listener is only enabled when TLS routing is enabled and does not enforce mTLS authentication since
+	// it's used to handle cluster join requests.
+	grpcInsecure net.Listener
```

Comment:
Can we name it differently. `Insecure` is often used in our code as a client TLS option when the client ignores server certs. 
What do you think about `grpcPublicListner` or  `grpcNoAuthListener`  ?